resources/components/ResourceTypeEditor.js

```diff
@@ -0,0 +1,253 @@
+import styles from '../styles/ResourceTypeEditor'
+
+const ResourceTypeEditor = compose(
+  connectFela(styles)
+)(props => {
+  const { resourceType = {} } = props
+  const { id = 'tmp' } = resourceType
+  const nextProps = merge(props, {
+    form: `resourceType-${id}`,
+    initialValues: resourceType
+  })
+  return h(ResourceTypeForm, nextProps)
+})
+
+export default ResourceTypeEditor
+
+const ResourceTypeForm = compose(
+  connectForm({}),
+)(props => {
+  const { styles, handleSubmit } = props
+
+  return h('form', {
+    className: styles.container,
+    onSubmit: handleSubmit
+  }, [
+    h(Field, {
+      name: 'name',
+      floatingLabelText: (
+        h(FormattedMessage, {
+          id: 'resources.resourceTypeName',
+          className: styles.labelText
+        })
+      ),
+      component: TextField
+    }),
+    h(ResourceTypeContains, props),
+    h(RaisedButton, {
+      type: 'submit',
+      className: styles.submitButton
+    }, [
+      h(FormattedMessage, {
+        id: 'resources.saveResourceType',
+        className: styles.buttonText
+      })
+    ])
+  ])
+})
+
+const ResourceTypeContains = compose(
+  // an atom is not reducible
+  // a tree is reducible (is composed of atoms)
+  withState('isReducible', 'setReducible', pipe(
+    propOr({}, 'initialValues'),
+    propOr([], 'items'),
+    length,
+    gte(__, 1)
+  ))
+)(props => {
+  const { styles, setReducible, isReducible, resourceTypes } = props
+
+  return h('div', {
+    className: styles.contains
+  }, [
+    h('div', {
+      className: styles.switchContainer
+    }, [
+      h(FormattedMessage, {
+        id: 'resources.isResourceTypeReducible',
+        className: styles.paragraphText
+      }),
+      h(RadioButtonGroup, {
+        name: 'isReducible',
+        onChange: (ev, value) => setReducible(value),
+        labelPosition: 'left',
+        defaultSelected: false,
+        value: isReducible
+      }, [
+        h(RadioButton, {
+          value: false,
+          label: (
+            h(FormattedMessage, {
+              id: 'resources.resourceTypeIsNotReducible',
+              className: styles.labelText
+            })
+          )
+        }),
+        h(RadioButton, {
+          value: true,
+          label: (
+            h(FormattedMessage, {
+              id: 'resources.resourceTypeIsReducible',
+              className: styles.labelText
+            })
+          ),
+        })
+      ])
+    ]),
+    isReducible && (
+      h(FieldArray, {
+        name: 'items',
+        component: ResourceTypeItemList,
+        styles,
+        resourceTypes
+      })
+    )
+  ])
+})
+
+const ResourceTypeItemList = (props) => {
+  const { styles, fields, resourceTypes } = props
+
+  return h('div', {
+    className: styles.itemListContainer
+  }, [
+    fields.map((field, index) => (
+      h(ResourceTypeItem, {
+        key: index,
+        styles,
+        field,
+        resourceTypes,
+        removeField: () => fields.remove(index)
```

Comment:
this doesn't look like it's working - trying to remove parts in storybook does nothing :(

Reply:
ah yeah, renamed from `removeItem` but didn't do below.